apps/nextjs/src/app/api/cron-jobs/clerk-user-cleanup/route.ts

```diff
@@ -0,0 +1,115 @@
+import { aiLogger } from "@oakai/logger";
+
+import { clerkClient } from "@clerk/nextjs/server";
+import type { User } from "@clerk/nextjs/server";
+import * as Sentry from "@sentry/node";
+import type { NextRequest } from "next/server";
+
+const log = aiLogger("cron");
+
+const DAY_IN_MS = 24 * 60 * 60 * 1000;
+const ONE_MONTH_MS = 30 * DAY_IN_MS;
+const THREE_MONTHS_MS = 90 * DAY_IN_MS;
+
+const OAK_EMAIL_WITH_NUMBER = /\d[^@]*@thenational\.academy$/;
+
+const shouldDeleteUser = (user: User): boolean => {
+  const email = user.primaryEmailAddress?.emailAddress;
+  if (!email) {
+    return false;
+  }
+
+  const lastActive = new Date(user.lastActiveAt ?? user.createdAt);
+  const created = new Date(user.createdAt);
+  const now = new Date();
+
+  const timeSinceLastActive = now.getTime() - lastActive.getTime();
+  const sameDay = lastActive.toDateString() === created.toDateString();
+
+  const hasTestCharacters =
+    email.includes("+") || OAK_EMAIL_WITH_NUMBER.test(email);
```

Comment:
I think some genuine Oak emails have numbers in them. I'm not sure if people use them but they are legitimate.

Reply:
Interesting insight, thank you. Would 3 numbers in a row rule those out?

Reply:
Yep sounds good. I've only seen the number '1' in the email, so two or more digits should be safe